src/Admin/ElectedRepresentativeAdmin.php

```diff
@@ -228,6 +243,25 @@ public function preSubmit(FormEvent $event): void
         }
     }
 
+    public function submit(FormEvent $event): void
+    {
+        /** @var ElectedRepresentative $electedRepresentative */
+        $electedRepresentative = $event->getData();
+
+        // if Adherent, we should add a LaREM label, if it does not exist
+        if ($electedRepresentative->isAdherent()
```

Comment:
as discussed, we need to remove the label if the adherent relation does not exist anymore.